migration/identity-migration/src/main/java/io/camunda/migration/identity/handler/sm/ClientMigrationHandler.java

```diff
@@ -46,57 +46,73 @@ protected List<Client> fetchBatch(final int page) {
   protected void process(final List<Client> batch) {
     final var clients = managementIdentityClient.fetchClients();
 
-    clients.forEach(
-        client -> {
-          final var clientId = client.clientId();
-          final var permissions =
-              managementIdentityClient.fetchClientPermissions(client.id()).stream()
-                  .map(
-                      permission ->
-                          String.format(
-                              "%s:%s",
-                              permission.resourceServerAudience(), permission.definition()))
-                  .toList();
-          if (permissions.isEmpty()) {
-            logger.debug(
-                "No permissions found for client '{}', skipping authorization creation.", clientId);
-            return;
-          }
-          totalAuthorizationsCount.incrementAndGet();
+    clients.stream()
+        .filter(
+            client -> {
+              switch (client.type()) {
+                case M2M, CONFIDENTIAL -> {
```

Comment:
I went for an inclusive solution to make sure, that in the unlikely event of new types we'll just skip them by default.